core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/claim/dao/ClaimDAO.java

```diff
@@ -407,6 +408,43 @@ protected void deleteClaimMapping(Connection dbConnection, String claimUri, Stri
     }
 
     /**
+     *
+     * @param dialectUri
+     * @throws UserStoreException
+     */
+    public void deleteDialectOnly(String dialectUri) throws UserStoreException {
+        Connection dbConnection = null;
+        PreparedStatement prepStmt = null;
+        try {
+            dbConnection = dataSource.getConnection();
+            dbConnection.setAutoCommit(false);
+            prepStmt = dbConnection.prepareStatement(ClaimDBConstants.DIALECT_CLAIM_MAPPINGS_EXIST);
+            prepStmt.setString(1, dialectUri);
+            prepStmt.setInt(2, tenantId);
+            prepStmt.setInt(3, tenantId);
+            ResultSet rs = prepStmt.executeQuery();
+            while (rs.next()) {
+                if (rs.getInt(1) == 0) {
+                    if (getDialect(dbConnection, dialectUri) != -1) {
+                        prepStmt = dbConnection.prepareStatement(ClaimDBConstants.DELETE_DIALECT);
+                        prepStmt.setString(1, dialectUri);
+                        prepStmt.setInt(2, tenantId);
+                        prepStmt.executeUpdate();
+                    }
+                }
+            }
+            prepStmt.close();
+
+            dbConnection.commit();
+        } catch (SQLException e) {
+            log.error("Database Error - " + e.getMessage(), e);
```

Comment:
no need to log e.getMessage(). it's logged with e. log some context information as the error message instead